apps/app-article/src/App.js

```diff
@@ -113,14 +125,15 @@ class App extends Component {
     }
 
     checkCache(uuid) {
-        if (JSON.parse(localStorage.getItem('cachedArticles')) != null) {
-            const articleFound = JSON.parse(localStorage.getItem('cachedArticles')).find(article => {
-                if (article.uuid === uuid) {
-                    return true;
-                }
-            });
-            return articleFound;
+        if(!isUserLoggedIn()){
+            if (JSON.parse(localStorage.getItem('cachedArticles')) != null) {
```

Comment:
We should first check if we find the article in the local storage, and then `JSON.parse` only one time.